mappings/loc.py

```diff
@@ -51,41 +59,49 @@ def applyFormatting(self):
     #Identifier/SourceID Formatting
     def formatIdentifierSourceID(self, itemList):
         newIdentifier = itemList
-        newIdentifier['call_number'][0] = f'{newIdentifier["call_number"][0]}|call_number'
         lccnNumber = self.record.identifiers[0][0]  #lccnNumber comes in as an array and we need the string inside the array
-        callNumber = newIdentifier['call_number'][0].strip(' ')
         sourceID = lccnNumber
+        if 'call_number' in newIdentifier.keys():
+            newIdentifier['call_number'][0] = f'{newIdentifier["call_number"][0]}|call_number'
+            callNumber = newIdentifier['call_number'][0].strip(' ')
+        else: 
+            callNumber = None
         return (lccnNumber, callNumber, sourceID)
     
     #Publisher/Spatial Formatting
     def formatPubSpatial(self, itemList):
         pubArray = []
-        spatialArray = []
-        for elem in itemList['created_published']:
-            if ':' not in elem:
-                createdPublishedList = elem.split(',', 1)
-                pubLocation = createdPublishedList[0].strip(' ')
-                if ',' in createdPublishedList[1]:
-                    pubOnly = createdPublishedList[1].split(',')[0].strip(' ')
+        spatialString = None
+        if 'created_published' in itemList.keys():
```

Comment:
more of just a nit, since it's functionally the same, but here, you don't need to call `.keys()` to check dictionary membership, you can just do `if 'created_published' in itemList.keys():`

Or if you wanna get fancy with the walrus operator, you can do:

```
if createdPublished := itemList.get('created_published', []):
    ...do stuff with `createdPublished
```

Up to you, just thought I'd throw it out there since there's a lot of instances in this file of checking dict membership and doing stuff with / iterating over values
```

Reply:
'but here, you don't need to call .keys() to check dictionary membership, you can just do if 'created_published' in itemList.keys():' The line you recommended is the same as what I wrote in the code. Did you mean to type something else? Also, the walrus operator is a interesting technique I'll look into using.